.circleci/config.yml

```diff
@@ -1,4 +1,5 @@
 version: 2
+executorType: machine
```

Comment:
I would like to consider the ramifications of this change and avoid it if we easily can. I know the Testcontainers documentation suggests to do this (https://www.testcontainers.org/supported_docker_environment/continuous_integration/circle_ci/), but I think this invalidates the docker images we are selecting to get a version of the JDK and it will take longer to start up since it is launching a VM.

I think we didn't change anything when running testcontainers with Zipkin on CircleCI, but I might be mistaken. Did you try without changing the executorType and got some error?

Reply:
Hmm maybe we were just skipping the tests with testcontainers on CircleCI in Zipkin projects... Looking at https://github.com/testcontainers/testcontainers-java/issues/1014 it sounds like running on the CircleCI 2.0 docker executors is still unsolved. This is unfortunate.

Reply:
@shakuzen It doesn't work for me without the change.

See the test result without "executorType: machine" (8d48e2b): https://314-111177506-gh.circle-artifacts.com/0/home/circleci/micrometer/test-results/junit/TEST-io.micrometer.elastic.ElasticsearchMeterRegistryElasticsearch5IntegrationTest.xml

and with "executorType: machine" (c04f963): https://315-111177506-gh.circle-artifacts.com/0/home/circleci/micrometer/test-results/junit/TEST-io.micrometer.elastic.ElasticsearchMeterRegistryElasticsearch5IntegrationTest.xml

I'm not sure there's a way to do this without the change ATM. We could give up running the tests in the CI although it sounds unusual.

Reply:
Perhaps trying DinD as the build image? https://hub.docker.com/_/docker/

Of course one would need to install the JDK himself, or we bake our own build-image.

Reply:
Avoiding needing to make our own images or deal with installing the JDK is the primary reason I like the docker executor. I'll try to find some alternatives.
I would definitely like the integration tests to run in the CI, though I guess if we can run them locally at least, that is better than now where we have no integration tests. Until we can figure out a more complete solution.

Reply:
It seems like the CircleCI way of achieving a similar thing would be to use the secondary images with different ES versions running, and get the host/port from environment variables. This might work, but it needlessly runs multiple ES images during the whole CI build and isn't portable to one's local environment.

Reply:
I will try to work with @bsideup to get testcontainers working on CircleCI's docker executors.